### ISSUE TITLE
Annotate functional interfaces with @FunctionalInterface

### ISSUE THREAD

**Issue by paulo-reichert** [GalateaRaj/fuse-starter-java#229]
Issue Description
--
Interfaces that are meant to be used as functional interfaces should be annotated with `@FunctionalInterface` to ensure that they have only one method and are clearly identifiable as such.

Two I can think right now are `ITranslator` and `IAgreementTransformer`.

Design
--
[Description of the root cause for the issue and how to resolve it. Should be complete before Review is approved]

Documentation Changes
--
[Description of changes necessary to the Fuse documentation due to the changes in response to this issue. Should be complete before Review is approved]

Test Evidence
--
![image](https://user-images.githubusercontent.com/43134061/45748443-a770a780-bc00-11e8-8b90-14911418a79e.png)

Validation in Develop
-- 
[Proof issue is resolved in Develop. Should be complete before Issue is closed]

**Comment by Gala-Sortino** [comment]
Already done with a pending pull request.

**Comment by GalateaEric** [comment]
PR: https://github.com/GalateaRaj/fuse-starter-java/pull/234 is blocked by #236
